pkg/parcacol/normalizer.go

```diff
@@ -234,7 +234,13 @@ type mappingNormalizationInfo struct {
 	offset int64
 }
 
-func (n *MetastoreNormalizer) NormalizeMappings(ctx context.Context, mappings []*pprofpb.Mapping, stringTable []string) ([]mappingNormalizationInfo, error) {
+// NormalizeMappings converts pprof mappings into the internal mapping format
+// and stores them in the metastore (Badger).
+func (n *MetastoreNormalizer) NormalizeMappings(ctx context.Context, mappings []*pprofpb.Mapping, stringTable []string, baseAddresses []uint64) ([]mappingNormalizationInfo, error) {
+	if len(baseAddresses) > len(mappings) {
```

Comment:
```suggestion
	if len(baseAddresses) != len(mappings) {
```
Maybe an equality check could be better to cover the case where `baseAddresses` is less than `mappings`.